src/Components/Publishing/Sections/Sections.tsx

```diff
@@ -11,79 +13,160 @@ import { Text } from "./Text"
 import { Video } from "./Video"
 
 interface SectionsProps {
+  DisplayPanel?: any
   article: {
     layout: Layout
     authors?: any
     postscript?: string
+    sections?: any
   }
+  isMobile?: boolean
 }
 
 interface StyledSectionsProps {
   layout: string
 }
 
-export const Sections: React.SFC<SectionsProps> = props => {
-  return (
-    <StyledSections layout={props.article.layout}>
-      {renderSections(props.article)}
-      {renderPostScript(props.article)}
-      {renderAuthors(props.article.authors)}
-    </StyledSections>
-  )
-}
+/**
+ * When isMobile, hide sidebar and inject DisplayAd into the body of the
+ * article at a specific paragraph index.
+ */
+const MOBILE_DISPLAY_PANEL_INJECT_INDEX = 2
+
+export class Sections extends React.Component<SectionsProps, any> {
+
+  injectDisplayPanelIntoText = once((body) => {
+    const displayPanelMarkup = ReactDOM.renderToString(<this.props.DisplayPanel />)
+
+    // Inject DisplayAd after a specific paragraph index
+    const tag = '</p>'
+    const updatedBody = body
+      .split(tag)
+      .map(p => p + tag)
+      .reduce((arr, block, paragraphIndex) => {
+        if (paragraphIndex === MOBILE_DISPLAY_PANEL_INJECT_INDEX) {
+          return arr.concat([block, displayPanelMarkup])
+        } else {
+          return arr.concat([block])
+        }
+      }, [])
+      .join('')
+
+    return updatedBody
+  })
+
+  getSection(section, layout) {
+    const sections = {
+      image_collection: (
+        <ImageCollection
+          sectionLayout={section.layout}
+          images={section.images}
+          targetHeight={500}
+          gutter={10}
+        />
+      ),
+      image_set: (
+        <ImageSetPreview
+          section={section}
+        />
+      ),
+      video: (
+        <Video
+          section={section}
+        />
+      ),
+      embed: (
+        <Embed
+          section={section}
+        />
+      ),
+      text: (
+        <Text
+          html={section.body}
+          layout={layout}
+        />
+      ),
+      default: false,
+    }
+
+    const sectionComponent = sections[section.type] || sections.default
+    return sectionComponent
+  }
+
+  renderSections() {
+    const { article, isMobile } = this.props
+    let displayAdInjected = false
+
+    const renderedSections = article.sections.map((sectionItem, i) => {
+      const shouldInject = !displayAdInjected && isMobile && sectionItem.type === 'text'
+      let section = sectionItem
+
+      if (shouldInject) {
+        displayAdInjected = true
+        section = clone(sectionItem)
+        section.body = this.injectDisplayPanelIntoText(section.body)
+      }
```

Comment:
I wonder if in the future we could simplify to targeting based on text sections rather than paragraphs. Interested to see how this QA's with various article types. cc @owendodd

Reply:
^  What I worry about. I can see this being kinda context sensitive, and was going to add a check to ensure that there even _are_ two paragraphs before inserting, otherwise keep moving on to the next text section